DuckDuckGoTests/NetworkProtectionInviteViewModelTests.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  NetworkProtectionInviteViewModelTests.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+@testable import DuckDuckGo
+import NetworkProtection
+import NetworkProtectionTestUtils
+
+final class NetworkProtectionInviteViewModelTests: XCTestCase {
+
+    func test_text_alwaysUppercased() {
+        let viewModel = viewModel()
+        viewModel.text = "abcdefg"
+        XCTAssertEqual(viewModel.text, "ABCDEFG")
+    }
+
+    func test_text_emptyString_disableSubmit() {
+        let viewModel = viewModel()
+        viewModel.text = ""
+        XCTAssertTrue(viewModel.shouldDisableSubmit)
+    }
+
+    func test_text_nonEmptyString_enableSubmit() {
+        let viewModel = viewModel()
+        for _ in 0..<5 {
+            viewModel.text.append("D")
+            XCTAssertFalse(viewModel.shouldDisableSubmit)
+        }
+    }
+
+    func test_submit_successfulRedemption_changesCurrentStepToSuccess() async {
+        let viewModel = viewModel(withInjectedRedemptionCoordinator: .whereRedeemSucceeds())
+        await viewModel.submit()
+        XCTAssertEqual(viewModel.currentStep, .success)
+    }
+
+    func test_submit_failedRedemption_doesNOTMoveToSuccess() async {
+        let viewModel = viewModel(withInjectedRedemptionCoordinator: .whereRedeemFails())
+        await viewModel.submit()
+        XCTAssertEqual(viewModel.currentStep, .codeEntry)
+    }
+
+    // Disabled this test but keeping it around to document the behaviour. It is failing inexplicably.
```

Comment:
For some reason, errors thrown in the library are not caught in the main app. This does not apply in production, only the test environment. You can manually test this to be sure by entering an incorrect invite code.